cask/src/cask/internal/VirtualThreadBlockingHandler.scala

```diff
@@ -0,0 +1,16 @@
+package cask.internal
+
+import io.undertow.server.{HttpHandler, HttpServerExchange}
+
+private[cask] final class VirtualThreadBlockingHandler(val handler: HttpHandler)
+  extends HttpHandler {
+  override def handleRequest(exchange: HttpServerExchange): Unit = {
+    exchange.startBlocking()
+    exchange.dispatch(VirtualThreadBlockingHandler.EXECUTOR, handler)
+  }
+}
+
+private[cask] object VirtualThreadBlockingHandler {
+  private lazy val EXECUTOR = new NewThreadPerTaskExecutor(
```

Comment:
Let's make this `Executor` live in the `Main` object, rather than global. That would let it be more easily configured separately for unit tests or other multi-server scenarios

Reply:
Make sense.

Reply:
I have a question about this:
When we expose the `executor` with `handlerExecutor` and then user can just override it with a `Executors.newThreadPerTaskExecutor` on Java 21, then the `MethodHandle` thing can be avoid.

I was expecting to support virtual thread out of box with `cask.virtualThread.enabled` if and only if this systeam property set to true and running with Java 21.

So I think the solution should be:
1. add `handlerExecutor` just as you suggested.
2. add a `cask.virtualThread.enabled` system property and run with Java 21 when user set that to true and running with Java 21, otherwise, without `2`, all the `MethodHandle` things seems unneeded.

I was busy on otherthings, sorry for the delay.